lib/load-session-from-socket.js

```diff
@@ -30,14 +30,18 @@ module.exports = function loadSessionFromSocket(socketHandshake, app, cb) {
 
   // If no cookie exists, generate one and set it on the handshake
   var hadToGenerateCookie;
-  if (!socketHandshake.headers.cookie){
-    socketHandshake.headers.cookie = app.session.generateNewSidCookie();
+  // Check if there is no cookie, or a cookie is present and it doesn't have sails.sid
+  if (!socketHandshake.headers.cookie || ('string' === typeof socketHandshake.headers.cookie &&
+    !~socketHandshake.headers.cookie.indexOf(app.config.session && app.config.session.name || 'sails.sid'))){
```

Comment:
Wondering if the presence of the cookie name keyword is enough and whether we should also include the `=` postfix?

Reply:
@shamasis in what situations do you think that we will not have the `=`? I was also tending towards but adding the `=` but couldn't think of a situation that would cause this

Reply:
Haha! I just don’t trust anything. That’s why.